modules/ts/tse/functions.cpp

```diff
@@ -0,0 +1,519 @@
+{
+	byte pid = params[1];
+	if (!IsPlayerValid(amx, pid)) return 0;
+	if (!Player(pid)->IsAlive()) return 0;
+	byte pwup = params[2];
+	float duration = amx_ctof(params[3]);
+	Player(pid)->SetPDataInt(453, pwup);
+	Player(pid)->SetPDataInt(452, pwup);
+	Player(pid)->SetPDataFloat(454, Player(pid)->GetTime() + duration);
+	return 1;
+}
+
+static cell AMX_NATIVE_CALL tse_setuserpwupduration(AMX *amx, cell *params)
+{
+	byte pid = params[1];
+	if (!IsPlayerValid(amx, pid)) return 0;
+	if (!Player(pid)->IsAlive()) return 0;
+	byte duration = params[2];
+	Player(pid)->SetPDataInt(455, duration);
+	return 1;
+}
+
+static cell AMX_NATIVE_CALL tse_configslowing(AMX *amx, cell *params)
+{
+	byte pid = params[1];
+	if (!IsPlayerValid(amx, pid)) return 0;
+	if (!Player(pid)->IsAlive()) return 0;
+	float speed = amx_ctof(params[2]);
+	float auradist = amx_ctof(params[3]);
+	Player(pid)->SetPDataFloat(90, speed);
+	Player(pid)->SetPDataFloat(91, auradist);
+	Player(pid)->SetPDataFloat(86, auradist);
+	return 1;
+}
+
+
+static cell AMX_NATIVE_CALL tse_configmeleeatk(AMX *amx, cell *params)
+{
+	byte pid = params[1];
+	if (!IsPlayerValid(amx, pid)) return 0;
+	if (!Player(pid)->IsAlive()) return 0;
+	float damage = amx_ctof(params[2]);
+	float duration = amx_ctof(params[3]);
+	Player(pid)->SetPDataFloat(483, damage);
+	Player(pid)->SetPDataFloat(482, duration);
+	return 1;
+}
+
+AMX_NATIVE_INFO pl_funcs[] = {
+	{ "tse_getuserslots", tse_getuserslots },
+	{ "tse_setuserslots", tse_setuserslots },
+	{ "tse_getusercash", tse_getusercash },
+	{ "tse_setusercash", tse_setusercash },
+	{ "tse_getuserkevlar", tse_getuserkevlar },
+	{ "tse_getuserstamina", tse_getuserstamina },
+	{ "tse_setuserstamina", tse_setuserstamina },
+	{ "tse_getuserstamina", tse_getuserstamina },
+	{ "tse_setuserstamina", tse_setuserstamina },
+	{ "tse_geteffectsphysspd", tse_geteffectsphysspd },
+	{ "tse_seteffectsphysspd", tse_seteffectsphysspd },
+	{ "tse_getuserphysspd", tse_getuserphysspd },
+	{ "tse_setuserphysspd", tse_setuserphysspd },
+	{ "tse_getusercurritems", tse_getusercurritems },
+	{ "tse_setusercurritems", tse_setusercurritems },
+	{ "tse_createpwup", tse_createpwup },
+	{ "tse_giveuserpwup", tse_giveuserpwup },
+	{ "tse_getuserpwup", tse_getuserpwup },
+	{ "tse_getuserstate", tse_getuserstate },
+	{ "tse_getuserstatus", tse_getuserstatus },
+	{ "tse_setuserstatus", tse_setuserstatus },
+	{ "tse_getuseractivepwup", tse_getuseractivepwup },
+	{ "tse_setfakepwup", tse_setfakepwup },
+	{ "tse_setuserpwupduration", tse_setuserpwupduration },
+	{ "tse_configslowing", tse_configslowing },
+	{ "tse_configmeleeatk", tse_configmeleeatk },
+	{ NULL, NULL }
+};
+
+static cell AMX_NATIVE_CALL tse_sendweapparams(AMX *amx, cell *params)
+{
+	byte pid = params[1];
+	if (!IsPlayerValid(amx, pid)) return 0;
+	char paramsstr[128];
+	sprintf(paramsstr, "%i %f %f %f %f %i", params[2], amx_ctof(params[3]), amx_ctof(params[4]), amx_ctof(params[5]), amx_ctof(params[6]), params[7]);
```

Comment:
I think it's better to use `snprintf` here, because buffer can overflow.

Reply:
Note: there is `<amtl/am-string.h>` for that, see `ke::SafeSprintf`.